src/main/cc/wfa/virtual_people/common/field_filter/equal_filter.cc

```diff
@@ -0,0 +1,203 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "wfa/virtual_people/common/field_filter/equal_filter.h"
+
+#include "absl/memory/memory.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_cat.h"
+#include "google/protobuf/descriptor.h"
+#include "google/protobuf/message.h"
+#include "src/main/proto/wfa/virtual_people/common/field_filter.pb.h"
+#include "wfa/virtual_people/common/field_filter/field_filter.h"
+#include "wfa/virtual_people/common/field_filter/utils/field_util.h"
+#include "wfa/virtual_people/common/field_filter/utils/type_convert_util.h"
+
+namespace wfa_virtual_people {
+
+// The implementation of EqualFilter based on the type of field represented by
+// @field_descriptors. The supported ValueTypes are
+//   int32_t
+//   int64_t
+//   uint32_t
+//   uint64_t
+//   bool
+//   const google::protobuf::EnumValueDescriptor*
+//   std::string
+template <typename ValueType>
+class EqualFilterImpl : public EqualFilter {
+ public:
+  explicit EqualFilterImpl(
+      std::vector<const google::protobuf::FieldDescriptor*>&& field_descriptors,
+      ValueType value):
+      field_descriptors_(std::move(field_descriptors)),
+      value_(value) {}
+
+  bool IsMatch(const google::protobuf::Message& message) const override;
+
+ private:
+  std::vector<const google::protobuf::FieldDescriptor*> field_descriptors_;
+  ValueType value_;
+};
+
+template <>
+class EqualFilterImpl<std::string> : public EqualFilter {
+ public:
+  explicit EqualFilterImpl(
+      std::vector<const google::protobuf::FieldDescriptor*>&& field_descriptors,
+      const std::string& value):
```

Comment:
Please use an absl::string_view for `value`.

Reply:
Done.